client/src/EntryForm.js

```diff
@@ -1,12 +1,12 @@
 import { useState } from 'react';
-import { addEntry, removeEntry, updateEntry } from './data';
+// import { addEntry, removeEntry, updateEntry } from './data';
 
 /**
  * Form that adds or edits an entry.
  * If `entry` is `null`, adds an entry.
  * If `entry` is defined, edits that entry.
  */
-export default function EntryForm({ entry, onSubmit }) {
+export default function EntryForm({ entry, onSubmit, entries, setEntries }) {
```

Comment:
You should name `setEntries` something that starts with `on`. Maybe `onSave`?

The solution I wrote puts the entries in EntryList instead of in App. That made it a bit easier to manage the events that modified the entries list.

My reasoning for putting it in EntryList is that the App may have multiple different things it is managing -- perhaps displaying entries, managing sign in/sign out, etc. So making it keep track of entries as well seems like it's asking it to do too much. EntryList, on the other hand, just displays and manages a list of entries, so having it retrieve them from the backend makes sense.

Your solution works and there's nothing wrong with it, it's just a slightly different approach than I took.